src/components/markdown-renderer/full-markdown-renderer.tsx

```diff
@@ -1,227 +1,61 @@
-import { replaceLegacyYoutubeShortCode } from './regex-plugins/replace-legacy-youtube-short-code'
-import { replacePdfShortCode } from './regex-plugins/replace-pdf-short-code'
-import { replaceQuoteExtraAuthor } from './regex-plugins/replace-quote-extra-author'
-import { replaceQuoteExtraColor } from './regex-plugins/replace-quote-extra-color'
-import { replaceQuoteExtraTime } from './regex-plugins/replace-quote-extra-time'
-import { replaceVimeoLink } from './regex-plugins/replace-vimeo-link'
-import { replaceYouTubeLink } from './regex-plugins/replace-youtube-link'
-import { AbcReplacer } from './replace-components/abc/abc-replacer'
-import { AsciinemaReplacer } from './replace-components/asciinema/asciinema-replacer'
-import { CsvReplacer } from './replace-components/csv/csv-replacer'
-import { FlowchartReplacer } from './replace-components/flow/flowchart-replacer'
-import { GistReplacer } from './replace-components/gist/gist-replacer'
-import { HighlightedCodeReplacer } from './replace-components/highlighted-fence/highlighted-fence-replacer'
-import { ImageReplacer } from './replace-components/image/image-replacer'
-import { KatexReplacer } from './replace-components/katex/katex-replacer'
 import { LinemarkerReplacer } from './replace-components/linemarker/linemarker-replacer'
-import { MermaidReplacer } from './replace-components/mermaid/mermaid-replacer'
-import { PdfReplacer } from './replace-components/pdf/pdf-replacer'
-import { PossibleWiderReplacer } from './replace-components/possible-wider/possible-wider-replacer'
-import { QuoteOptionsReplacer } from './replace-components/quote-options/quote-options-replacer'
-import { SequenceDiagramReplacer } from './replace-components/sequence-diagram/sequence-diagram-replacer'
-import { TaskListReplacer } from './replace-components/task-list/task-list-replacer'
-import { VimeoReplacer } from './replace-components/vimeo/vimeo-replacer'
-import { YoutubeReplacer } from './replace-components/youtube/youtube-replacer'
-import { useCalculateLineMarkerPosition } from './utils/calculate-line-marker-positions'
 import { AdditionalMarkdownRendererProps, LineMarkerPosition } from './types'
-import { usePostMetaDataOnChange } from './utils/use-post-meta-data-on-change'
-import { usePostTocAstOnChange } from './utils/use-post-toc-ast-on-change'
+import { useCalculateLineMarkerPosition } from './utils/calculate-line-marker-positions'
 
 export interface FullMarkdownRendererProps {
-  onFirstHeadingChange?: (firstHeading: string | undefined) => void
   onLineMarkerPositionChanged?: (lineMarkerPosition: LineMarkerPosition[]) => void
+  onFirstHeadingChange?: (firstHeading: string | undefined) => void
   onMetaDataChange?: (yamlMetaData: YAMLMetaData | undefined) => void
   onTaskCheckedChange: (lineInMarkdown: number, checked: boolean) => void
   onTocChange?: (ast: TocAst) => void
 }
 
 export const FullMarkdownRenderer: React.FC<FullMarkdownRendererProps & AdditionalMarkdownRendererProps> = ({
```

Comment:
what is the difference between the full markdown renderer and the enhanced one?..

Reply:
The enhanced renderer extends the basic renderer by enabling diagrams, one-click-components and all the other fancy stuff of HedgeDoc.

The full renderer extends the enhanced renderer by handling the line-number-markers and adding the line-marker replacer.